protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftServiceManager.java

```diff
@@ -245,7 +247,9 @@ public long getCompactableTerm() {
     ComposableFuture<Snapshot> future = new ComposableFuture<>();
     stateContext.execute(() -> {
       try {
+        long startTime = System.currentTimeMillis();
```

Comment:
I think `System.nanoTime()` is typically the recommended way of measuring elapsed time, as `currentTimeMillis` returns wall-clock time, whereas `nanoTime` is unrelated to actual time but refers only to elapsed time since some unknown origin. Guava's `Stopwatch` is another option which basically uses `nanoTime` but smooths out any gotchas associated with it.

I'm no Java expert though :sweat_smile: this is just what I've gotten from reading about it so far

Reply:
Isn't `System.nanoTime()` more expensive? Since we are not interested in very precise measurement, I thought `currentTimeMillis` would be ok.

Reply:
I was under the impression it was the other way around, though now I'm not sure :sweat_smile: